src/language_server_protocol.rs

```diff
@@ -3548,23 +3548,21 @@ impl LanguageClient {
         info!("Begin {}", REQUEST__ExplainErrorAtPoint);
         let filename = self.vim()?.get_filename(params)?;
         let position = self.vim()?.get_position(params)?;
-        let diag = self.get(|state| {
-            state
-                .diagnostics
-                .get(&filename)
-                .ok_or_else(|| format_err!("No diagnostics found: filename: {}", filename,))?
-                .iter()
-                .find(|dn| position >= dn.range.start && position < dn.range.end)
-                .cloned()
-                .ok_or_else(|| {
-                    format_err!(
-                        "No diagnostics found: filename: {}, line: {}, character: {}",
-                        filename,
-                        position.line,
-                        position.character
-                    )
-                })
-        })??;
+        let diag = self
+            .get(|state| state.diagnostics.get(&filename).cloned())?
+            .unwrap_or_default()
+            .iter()
+            .find(|dn| position >= dn.range.start && position < dn.range.end)
+            .cloned();
+
+        if diag.is_none() {
+            debug!(
```

Comment:
May I suggest adding a user config and default to the old behaviour here?

The reason is because, when user calls this method explicitly, something is expected from user end. If we did nothing here, user might be confused or wondering why the request hasn't fulfilled yet.

Make sense?

Reply:
What we could do is add structured logging (on the vim side) and add a config to filter the messages based on the level, as to avoid having a config value for a specific error message, but that sounds like a little more work so maybe we hold that off until we find that it's really needed?

Meanwhile, I'm ok with not merging this if you think it may be confusing for users.